app/javascript/components/workflow-credential-mapping-form/index.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import { CredentialMapperComponent } from './helper';
+import componentMapper from '../../forms/mappers/componentMapper';
+import createSchema from './workflow-credential-mapping-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const WorkflowCredentialMappingForm = ({ recordId }) => {
+  const [{
+    credentials,
+    credentialReferences,
+    payloadCredentials,
+    workflowAuthentications,
+    initialValues,
+    isLoading,
+  }, setState] = useState({ isLoading: !!recordId });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  // custom component mapper
+  const mapper = {
+    ...componentMapper,
+    'credential-mapper': CredentialMapperComponent,
+  };
+
+  useEffect(() => {
+    // eslint-disable-next-line camelcase
+    API.get(
+      `/api/authentications?expand=resources&filter[]=type='ManageIQ::Providers::Workflows::AutomationManager::WorkflowCredential'`
+    ).then(({ resources }) => {
+      API.get(`/api/configuration_script_payloads/${recordId}`).then(({ name, payload, credentials }) => {
+        const initialCredentials = credentials != null ? credentials : {};
+
+        /*
+          Creates the list of credential references from the workflow payload by parsing each state and saving them to payloadCredentials.
+          Duplicate references get overridden.
+        */
+        const jsonPayload = JSON.parse(payload).States;
+        let payloadCredentials = {};
+
+        Object.keys(jsonPayload).forEach((key1) => {
+          if (jsonPayload[key1].Credentials != null) {
+            Object.keys(jsonPayload[key1].Credentials).forEach((key2) => {
+              payloadCredentials = {
+                [key2]: jsonPayload[key1].Credentials[key2],
+                ...payloadCredentials,
+              };
+            });
+          }
+        });
```

Comment:
@agrare Discussed this with @DavidResende0 and while I'm ok with this for this PR, it seems weird to me that the UI would have to parse the payload since that's really the backend's job.  I'm wondering if we should extract any credentials into a separate column, so they can be more easily accessed here.  Thoughts?

Reply:
One way we could do this is pre-parse the payload and store those in the credentials column, but with a `nil` value.  That way, this form would only show the keys and if those keys have non-nil values, then they must be mapped.

Reply:
Yeah I agree, this was actually something I discussed with @DavidResende0 a few weeks ago but I never got to implementing, I was thinking that when we load the payload and create the record we could pre-populate the credentials record with the keys that have to be filled in by the user

Reply:
Ha just edged me out